### DISCUSSION TITLE
Sanitize SVG inside regular HTML

### DISCUSSION THREAD

**Discussion by sonnatager** [mganss/HtmlSanitizer#555] [General]
Hello guys,

I have following piece of HTML:
```html
<div>
   <a href="/any/link">This is a test link</a>
   <div style="font-size: 30em;">
      <svg viewBox="0 0 30 10" xmlns="http://www.w3.org/2000/svg">
         <circle id="myCircle" cx="5" cy="5" r="4" stroke="blue" />
         <use href="#myCircle" x="10" fill="blue" />
         <use href="#myCircle" x="20" fill="white" stroke="red" />
      </svg>
   </div>
</div>
```

I have now added [this](https://github.com/darylldoyle/svg-sanitizer/blob/master/src/data/AllowedAttributes.php)  attributes and [this](https://github.com/darylldoyle/svg-sanitizer/blob/master/src/data/AllowedTags.php)  tags to the allowedAttributes and allowedTags array of the sanitizer.

My question is, how can i configure the sanitizer to use the added allowed svg tags and attributs just for the svg element in the html. The html "outside" the svg (e.g. the <a> tag) should just use the default allowed tags and attributes.

**Comment by mganss** [comment]
My suggestion is to not add the additional tags and attributes to `AllowedTags` and `AllowedElements` and instead use events to check if an element is a child of an svg element, leveraging the AngleSharp API. Here's more about the available events and how to use them: https://github.com/mganss/HtmlSanitizer/wiki/Hooks

**Comment by sonnatager** [comment]
Ah yes, i didn't thought about that possibility :-)

I will try that
